File: src/2022/day03/src/main/java/ca/terrylockett/aoc2022/day03/Day03Runner.java
package ca.terrylockett.aoc2022.day03;

import ca.terrylockett.aoccommon.inputfilefinder.Resources;

public class Day03Runner {

	final static String INPUT_FILE_NAME = "input.txt";

	public static void main(String[] args) throws Exception {

		String inputFilePath = Resources.getInputFilePath(INPUT_FILE_NAME).orElseThrow();

		// ******** Part 01 ********
		System.out.println("Day3 Part1: " + RuckSackValidator.findDuplicateItemsTotal(inputFilePath));

		// ******** Part 02 ********
		System.out.println("Day3 Part2: " + RuckSackValidator.findBadgeTotals(inputFilePath));
	}

}

File: src/2023/day03/src/main/kotlin/ca/terrylockett/aoc2023/day03/Day03Runner.kt
package ca.terrylockett.aoc2023.day03

import ca.terrylockett.aoccommon.resources.Resources

fun main() {
	val inputFile: String = Resources.getInputFilePath("input.txt").orElseThrow()
	val engine = EngineSchematic(inputFile)

	println("2023 day03 part1: ${engine.part1FindParts()}")
	println("2023 day03 part2: ${engine.part2FindGears()}")
}
